share/completions/zypper.fish

```diff
@@ -29,12 +31,50 @@ function __fish_zypper_use_repo
 end
 
 function __fish_zypper_print_repos
-    for repofile in /etc/zypp/repos.d/*.repo
-        string replace -f -r '\[(.+)\]' '$1' < $repofile
+    zypper -t lr | sed '1,/-+-/d' | cut -d '|' -f 2 | cut -d ' ' -f 2
+end
+
+function __fish_zypper_print_enabled_repos
+    zypper -t lr -E | sed '1,/-+-/d' | cut -d '|' -f 2 | cut -d ' ' -f 2
+end
+
+function __fish_zypper_print_packages
+    set -l repo
+    set -l type
+    set -l idx
+    set -l args (commandline -poc)
+    while test -n "$args"
```

Comment:
You probably want `while set -q args[1]` instead. Otherwise it'll stop if there is an empty argument (like `""`) in the middle.